web/controllers/auth_controller.ex

```diff
@@ -0,0 +1,46 @@
+defmodule CodeCorps.AuthController do
+  use CodeCorps.Web, :controller
+  import Comeonin.Bcrypt, only: [checkpw: 2, dummy_checkpw: 0]
+  alias CodeCorps.Repo
+  alias CodeCorps.User
+
+  def create(conn, params = %{"email" => _, "password" => _}) do
+    case login_by_email_and_pass(conn, params) do
+      {:ok, user} ->
+        {:ok, jwt, _full_claims} = user |> Guardian.encode_and_sign(:token)
+
+        conn
+        |> put_status(:created)
+        |> render("show.json", jwt: jwt)
+
+      _ ->
+        conn
+        |> put_status(:unprocessable_entity)
+        |> render("error.json")
+    end
+  end
+
+  def delete(conn, _params) do
+    {:ok, claims} = Guardian.Plug.claims(conn)
+
+    conn
+    |> Guardian.Plug.current_token
+    |> Guardian.revoke!(claims)
+
+    conn
+    |> render("delete.json")
+  end
+
+  defp login_by_email_and_pass(conn, %{"email" => email, "password" => password}) do
+    user = Repo.get_by(User, email: email)
+    cond do
+      user && checkpw(password, user.encrypted_password) ->
+        {:ok, user}
+      user ->
+        {:error, :unauthorized}
+      true ->
+        dummy_checkpw()
+        {:error, :not_found}
+    end
```

Comment:
This could maybe be reworked without a condition. 

```Elixir
defp login_by_email_and_pass(conn, %{"email" => email, "password" => password}) do
  User
  |> Repo.get_by(email: email)
  |> do_login_by_email_and_pass(password)
end

defp do_login_by_email_and_pass(nil, _) do
  dummy_checkpw()
  {:error, :not_found}
end

defp do_login_by_email_and_pass(user, password) do
  checkpw(password, user.encrypted_password)
 |> process_login_attempt(user)
end

defp process_login_attempt(true, user), do: {:ok, user}
defp process_login_attempt(false, _), do: {:error, :unauthorized}
```

I know the code is larger, but it should be more elixir-like.